### PR TITLE
Update docker instructions to use bash

### PR DESCRIPTION
**Pull request by hamishwillee**
@julianoes This updates the instructions for building using Docker.  The main changes are that:
-  this downloads the image rather than building it in the main example. This is usually a lot faster. I still show how to build the image in a later section
- This opens a bash prompt. Main reason is that if you want to build the application then this is much easier in bash - and so are instructions: "just pretend you are on linux" :-)
- I do also show the same things you did, as sub sections.

### CODE REVIEW

**Comment by hamishwillee** [comment]
So do I need to clean things - what if I don't? Is the implication that I just can't access the files?
Can I make myself the owner - ie pass current user into docker for the build?
If I run sudo make clean outside of docker, do I need anything special in my environment

**Comment by julianoes** [comment]
> So do I need to clean things - what if I don't? Is the implication that I just can't access the files?

You just can't clean it as a user.

>  Can I make myself the owner - ie pass current user into docker for the build?

It's a bit complicated. There is the way how it is done for PX4 containers:
https://github.com/PX4/containers/blob/master/docker/px4-dev/scripts/entrypoint.sh
I want to try this some time but it's bit more work, so not immediately.

>  If I run sudo make clean outside of docker, do I need anything special in my environment

No.

File: en/contributing/build.md [hunk 1]
```diff
@@ -185,23 +185,75 @@ build/default/integration_tests_runner --gtest_filter="ExternalExampleHello"
...
+   make default install
+   # Run code-style check
+   make fix_style
+   # Build the docs
+   make docs
+   # Clean the build
+   make clean
+   ```
+   
+> **Note** Files built in a Docker container are owned by root. In order to clean up the **build** and **install** folders you will need to either call `make clean` in the container or `sudo make clean` in the host computer.
```
Comment (hamishwillee) [inline en/contributing/build.md]: Thanks for clarification @julianoes . I added this note and merged.

Comment (julianoes) [reply en/contributing/build.md]: Great thanks.

**Pull request merged**